src/DataTypes/Serializations/SerializationEnum.h

```diff
@@ -35,8 +47,12 @@ class SerializationEnum : public SerializationNumber<Type>, public EnumValues<Ty
     {
         FieldType x;
         readText(x, istr);
-        return this->findByValue(x)->first;
+        return ref_enum_values->findByValue(x)->first;
     }
+
+    std::optional<EnumValues<Type>> own_enum_values;
+    std::shared_ptr<const DataTypeEnum<Type>> own_enum_type;
+    const EnumValues<Type> * ref_enum_values;
```

Comment:
It would actually be nicer if this could be a const reference (and thus be immutable + have a bit nicer syntax) and additionally be renamed to just `enum_values`.